crates/bevy_render/src/camera/camera.rs

```diff
@@ -66,6 +66,31 @@ impl Default for Viewport {
     }
 }
 
+/// Settings to define a camera sub view.
+///
+/// When [`Camera::sub_camera_view`] is `Some`, only the sub-section of the
+/// image defined by `size` and `offset` (relative to the `full_size` of the
+/// whole image) is projected to the cameras viewport.
+#[derive(Debug, Clone, Copy, Reflect, PartialEq, Eq)]
+pub struct SubCameraView {
+    /// Size of the entire camera view
+    pub full_size: UVec2,
```

Comment:
I'm a bit confused about the purpose of this field, shouldn't this be computed from the camera/viewport itself rather than manually specified?

Reply:
It could theoretically be calculated from the viewport itself, however I think this would make the API more confusing for users. Take the example of a multi-monitor setup for example.

```
┌───┬───┐
│ A │ B │
├───┼───┤
│ C │ D │
└───┴───┘
```
If each monitor is 1920x1080 and we decide to use a 1920x1080 viewport for each monitor, then the whole should be 3840x2160.

Now to determine the sub view values for monitor/viewport A, the sub view has to have half the dimensions of the "full image":
- If we calculated the full view from the size of our viewport, the sub view would have to be 960x540 to correctly take up a quarter of the physical viewport that is 1920x1080
- If we have the additional `full_size` field, the users don't have to think about the actual size of their physical viewport but can consider the size of the "full image" instead. And can then just use the value of the physical viewport of A for their sub view.

As I write this out I'm starting to realize that this should also work with `full_size` being 32x18 and `size` being 16x9.

I will change the example to make this more apparent, and adjust the doc comments to make it clearer that the ratio is what is relevant. 

I will also allow for offset to be a regular `Vec2` to allow for smaller steps when `size` and `full_size` is made up of small values.

Reply:
Alright, that makes sense to me.